### PR TITLE
[Woo POS][Cash & Receipts] Order completion call. Navigation to the payment success

### LINKED ISSUE #13084: [Woo POS][Cash & Receipts] Implement order completion call during cash payment flow with navigation to the payment success

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #13084

### Testing information

* More -> POS -> New order
* Choose pay with cash
* Complete the order
* Notice that success screen is shown
* Go to the order list in the main app. Notice new, complete order. Check it detals

Try back navigation from the cash payment screen, too

### The tests that have been performed

Above

### Images/gif

https://github.com/user-attachments/assets/346bb788-d5e9-4100-b5c2-af3695da9d04

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentRepository.kt [hunk 1]
```diff
@@ -1,20 +1,61 @@
...
+        ) ?: WCOrderStatusModel(statusKey = Order.Status.Completed.value).apply {
+            label = statusKey
+        }
+
+        orderStore.updateOrderStatusAndPaymentMethod(
+            orderId = orderId,
+            site = selectedSite.get(),
+            newStatus = statusModel,
+            newPaymentMethodId = CASH_ON_DELIVERY_PAYMENT_TYPE,
+            codGateway?.title ?: "Pay in Person",
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentRepository.kt]:
```suggestion
            newPaymentMethodTitle = codGateway?.title ?: "Pay in Person",
```
💡 np

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentScreen.kt [hunk 2]
```diff
@@ -196,11 +199,15 @@ fun WooPosTotalsPaymentCashScreenScreen() {
                 enteredAmount = "5$",
                 changeDue = "5$",
                 total = "10$",
-                canBeOrderBeCompleted = true,
+                button = WooPosCashPaymentState.Collecting.Button(
+                    text = "Complete order",
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentScreen.kt:203]: 💡 To Res

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentViewModel.kt [hunk 3]
```diff
@@ -34,15 +37,42 @@ class WooPosCashPaymentViewModel @Inject constructor(
...
         when (event) {
             is WooPosCashPaymentUIEvent.AmountChanged -> TODO()
-            WooPosCashPaymentUIEvent.CompleteOrderClicked -> TODO()
+            WooPosCashPaymentUIEvent.CompleteOrderClicked -> handleOrderCompletion()
+        }
+    }
+
+    private fun handleOrderCompletion() {
+        viewModelScope.launch {
+            val stateBeforeCompleting = _state.value as? WooPosCashPaymentState.Collecting ?: return@launch
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentViewModel.kt]: ❓ Do we expect other states – shouldn't we crash? What will happen if we return here?

**Review by samiuelson — APPROVED** [review]
Looks good, @kidinov. I left some non-blocking comments. 

I also noticed that the Totals screen is visible for a while before Payment successful is shown. Can we avoid that? See video:

https://github.com/user-attachments/assets/b9f32edf-abcd-4e99-870c-3ec68ee7c192

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentScreen.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentScreen.kt:203]: Yep, I implemented the UI in the following PR

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentViewModel.kt (hunk 3 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/cashpayment/WooPosCashPaymentViewModel.kt]: Good call. Change this case to strong casting. After the call though, kept it as it is as we don't know if the state changed during the network call duration 

568743fb300f6cea70f2dbf622a59c053aff9fd1

**Comment by kidinov** [comment]
@samiuelson 

> I also noticed that the Totals screen is visible for a while before Payment successful is shown. Can we avoid that? See video:

Yep, I already validated this with other people and it indeed looks weird. I have a ticket to redo that

**Pull request merged**